alica_engine/include/engine/PlanBase.h

```diff
@@ -14,91 +14,93 @@
-		Plan* getMasterPlan();
+    /**
+     * A PlanBase holds the internal representation of the plan graph and issues all operations on it.
+     * It is the most central object within the ALICA Engine.
+     */
+    class PlanBase
+    {
+    public:
+        PlanBase(AlicaEngine* ae, Plan* masterplan);
+        ~PlanBase();
+        std::condition_variable* getStepModeCV();
+        const std::shared_ptr<RunningPlan> getRootNode() const;
+        void setRootNode(std::shared_ptr<RunningPlan> rootNode);
+        const AlicaTime getloopInterval() const;
+        void setLoopInterval(ulong loopInterval);
+        void stop();
+        void start();
+        void addFastPathEvent(shared_ptr<RunningPlan> p);
+        std::shared_ptr<RunningPlan> getDeepestNode();
+        std::shared_ptr<RunningPlan> getRootNode();
 
-	private:
-		/**
-		 * List of RunningPlans scheduled for out-of-loop evaluation.
-		 */
-		queue<shared_ptr<RunningPlan>> fpEvents;
-		condition_variable* stepModeCV;
-//		void checkPlanBase(shared_ptr<RunningPlan> r);
+        /*const*/ Plan* getMasterPlan() const {return _masterPlan;}
+        bool isWaiting() const { return _isWaiting; }
+    private:
+        void run();
 
-	protected:
-		Plan* masterPlan;
-		shared_ptr<RunningPlan> rootNode;
-		shared_ptr<RunningPlan> deepestNode;
-		AlicaEngine* ae;
-		int treeDepth;
-		RuleBook* ruleBook;
-		TeamObserver* teamObserver;
-		IRoleAssignment* ra;
-		SyncModule* syncModel;
-		AuthorityManager* authModul;
-		IAlicaCommunication* statusPublisher;
-		IAlicaClock* alicaClock;
+        /**
+         * List of RunningPlans scheduled for out-of-loop evaluation.
+         */
 
-		AlicaTime loopTime;
-		AlicaTime lastSendTime;
-		AlicaTime minSendInterval;
-		AlicaTime maxSendInterval;
-		AlicaTime loopInterval;
-		AlicaTime lastSentStatusTime;
-		AlicaTime sendStatusInterval;
+        AlicaEngine* _ae;
+        Plan* _masterPlan;
+        
+        TeamObserver* _teamObserver;
+        IRoleAssignment* _ra;
+        SyncModule* _syncModel;
+        AuthorityManager* _authModul;
+        IAlicaCommunication* _statusPublisher;
+        IAlicaClock* _alicaClock;
 
-		bool running;
-		bool sendStatusMessages;
+        shared_ptr<RunningPlan> _rootNode;
+        shared_ptr<RunningPlan> _deepestNode;
 
-		thread* mainThread;
-		Logger* log;
+        thread* _mainThread;
+        Logger* _log;
+        AlicaEngineInfo* _statusMessage;
 
-		AlicaEngineInfo* statusMessage;
-		mutex lomutex;
-		mutex stepMutex;
-		void run();
+        AlicaTime _loopTime;
+        AlicaTime _lastSendTime;
+        AlicaTime _minSendInterval;
+        AlicaTime _maxSendInterval;
+        AlicaTime _loopInterval;
+        AlicaTime _lastSentStatusTime;
+        AlicaTime _sendStatusInterval;
 
+        std::mutex _lomutex;
+        std::mutex _stepMutex;
 
-	};
+        queue<shared_ptr<RunningPlan>> _fpEvents;
+        std::condition_variable _stepModeCV;
+        RuleBook _ruleBook;
```

Comment:
Why no pointer anymore?

Reply:
No need to be a pointer. The less memory indirections we have the better.